File: modules/expo-media-library/android/src/main/java/expo/modules/medialibrary/DeleteAssets.java
package expo.modules.medialibrary;

import android.content.Context;
import android.os.AsyncTask;
import android.provider.MediaStore;
import android.text.TextUtils;

import expo.core.Promise;

import static expo.modules.medialibrary.MediaLibraryUtils.deleteAssets;

class DeleteAssets extends AsyncTask<Void, Void, Void> {

  private final Context mContext;
  private final String[] mAssetsId;
  private final Promise mPromise;

  DeleteAssets(Context context, String[] assetsId, Promise promise) {
    mContext = context;
    mAssetsId = assetsId;
    mPromise = promise;
  }

  @Override
  protected Void doInBackground(Void... params) {
    final String selection = MediaStore.Images.Media._ID + " IN (" + TextUtils.join(",", mAssetsId) + " )";
    final String[] selectionArgs = null;
    deleteAssets(mContext, selection, selectionArgs, mPromise);
    return null;
  }
}

File: packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/assets/DeleteAssets.kt
package expo.modules.medialibrary.assets

import android.content.Context
import android.provider.MediaStore
import expo.modules.medialibrary.MediaLibraryUtils

suspend fun deleteAssets(context: Context, assetIds: Array<String>): Boolean {
  val selection = "${MediaStore.Images.Media._ID} IN (${assetIds.joinToString(separator = ",")} )"
  val selectionArgs: Array<String>? = null

  return MediaLibraryUtils.deleteAssets(context, selection, selectionArgs)
}
